SSA/Core/Framework.lean

```diff
@@ -15,19 +15,24 @@ open TyDenote (toType)
   # Classes
 -/
 
+/-- A MLIR `Dialect` is comprised of a type of `Op`erations, and a type of `Ty`pes -/
+structure Dialect where
+  Op : Type
+  Ty : Type
+
 abbrev RegionSignature Ty := List (Ctxt Ty × Ty)
 
 structure Signature (Ty : Type) where
   sig     : List Ty
   regSig  : RegionSignature Ty
   outTy   : Ty
 
-class OpSignature (Op : Type) (Ty : outParam (Type)) where
-  signature : Op → Signature Ty
+class OpSignature (d : Dialect) where
```

Comment:
Shouldn't this be `DialectSignature` then?

Reply:
I've also renamed `OpDenote` -> `DialectDenote`